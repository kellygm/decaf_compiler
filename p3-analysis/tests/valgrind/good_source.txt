==2557612== Memcheck, a memory error detector
==2557612== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==2557612== Using Valgrind-3.22.0 and LibVEX; rerun with -h for copyright info
==2557612== Command: ../decaf inputs/good.decaf
==2557612== 
Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 main : () -> int
 i : int
 b : bool

  FuncDecl name="main" return_type=int parameters={} [line 3]
  SYM TABLE:

    Block [line 4]
    SYM TABLE:
     j : int
     c : bool

==2557612== 
==2557612== HEAP SUMMARY:
==2557612==     in use at exit: 0 bytes in 0 blocks
==2557612==   total heap usage: 1,902 allocs, 1,902 frees, 222,960 bytes allocated
==2557612== 
==2557612== All heap blocks were freed -- no leaks are possible
==2557612== 
==2557612== For lists of detected and suppressed errors, rerun with: -s
==2557612== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
