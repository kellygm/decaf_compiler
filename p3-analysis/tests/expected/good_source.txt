Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 main : () -> int
 i : int
 b : bool

  FuncDecl name="main" return_type=int parameters={} [line 3]
  SYM TABLE:

    Block [line 4]
    SYM TABLE:
     j : int
     c : bool

